hikari/impl/rest.py

```diff
@@ -4377,3 +4392,88 @@ def fetch_scheduled_event_users(
+        response = await self._request(routes.GET_APPLICATION_SKUS.compile(application=str(int(application))))
+        assert isinstance(response, list)
+
+        return [self._entity_factory.deserialize_sku(payload) for payload in response]
+
+    async def fetch_entitlements(
+        self,
+        application: snowflakes.SnowflakeishOr[guilds.PartialApplication],
+        /,
+        *,
+        user: undefined.UndefinedOr[snowflakes.SnowflakeishOr[users.PartialUser]] = undefined.UNDEFINED,
+        guild: undefined.UndefinedOr[snowflakes.SnowflakeishOr[guilds.PartialGuild]] = undefined.UNDEFINED,
+        before: undefined.UndefinedOr[snowflakes.SearchableSnowflakeish] = undefined.UNDEFINED,
+        after: undefined.UndefinedOr[snowflakes.SearchableSnowflakeish] = undefined.UNDEFINED,
+        limit: undefined.UndefinedOr[int] = undefined.UNDEFINED,
+        exclude_ended: undefined.UndefinedOr[bool] = undefined.UNDEFINED,
+    ) -> typing.Sequence[monetization.Entitlement]:
+        query = data_binding.StringMapBuilder()
+        if user is not undefined.UNDEFINED:
+            query.put("user_id", str(int(user)))
+
+        if guild is not undefined.UNDEFINED:
+            query.put("guild_id", str(int(guild)))
+
+        if before is not undefined.UNDEFINED:
+            if isinstance(before, datetime.datetime):
+                query.put("before", str(snowflakes.Snowflake.from_datetime(before)))
+            else:
+                query.put("before", str(int(before)))
+
+        if after is not undefined.UNDEFINED:
+            if isinstance(after, datetime.datetime):
+                query.put("after", str(snowflakes.Snowflake.from_datetime(after)))
+            else:
+                query.put("after", str(int(after)))
+
+        if limit is not undefined.UNDEFINED:
+            query.put("limit", limit)
+
+        if exclude_ended is not undefined.UNDEFINED:
+            query.put("exclude_ended", exclude_ended)
+
+        response = await self._request(
+            routes.GET_APPLICATION_ENTITLEMENTS.compile(application=str(int(application))), query=query
+        )
+        assert isinstance(response, list)
+
+        return [self._entity_factory.deserialize_entitlement(payload) for payload in response]
+
+    async def create_test_entitlement(
+        self,
+        application: snowflakes.SnowflakeishOr[guilds.PartialApplication],
+        /,
+        *,
+        sku: snowflakes.SnowflakeishOr[monetization.SKU],
+        owner_id: snowflakes.Snowflakeish,
+        owner_type: monetization.EntitlementOwnerType,
+    ) -> monetization.Entitlement:
+        body = data_binding.JSONObjectBuilder()
+        body.put("sku_id", str(int(sku)))
+        body.put("owner_id", str(int(owner_id)))
+        body.put("owner_type", int(owner_type))
+
+        response = await self._request(
+            routes.POST_APPLICATION_TEST_ENTITLEMENT.compile(application=str(int(application))), json=body
+        )
```

Comment:
It would also be nice to change the logic to how `edit_permission_overwrite` is implemented to not require passing `owner_type` if we can figure out what it is, not that i think this route will be used a lot

```suggestion
        body = data_binding.JSONObjectBuilder()
        body.put("sku_id", sku)
        body.put("owner_id", owner)
        body.put("owner_type", owner_type)

        route = routes.POST_APPLICATION_TEST_ENTITLEMENT.compile(application=application)
        response = await self.request(route, json=body)
```